packages/api/tests/endpoints/async-operations/test-create-async-operations.js

```diff
@@ -0,0 +1,480 @@
+
+  t.is(asyncOperationPgRecord.created_at.getTime(), record.createdAt);
+  t.is(asyncOperationPgRecord.updated_at.getTime(), record.updatedAt);
+  t.is(asyncOperationPgRecord.created_at.getTime(), esRecord.createdAt);
+  t.is(asyncOperationPgRecord.updated_at.getTime(), esRecord.updatedAt);
+});
+
+test('POST returns a 409 error if the async operation already exists in DynamoDB', async (t) => {
+  const { asyncOperationModel, jwtAuthToken } = t.context;
+  const asyncOperation = {
+    id: uuidv4(),
+    status: 'RUNNING',
+    taskArn: randomString(),
+    description: 'Some async run',
+    operationType: 'Bulk Granules',
+    output: JSON.stringify({ age: 59 }),
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  };
+
+  await asyncOperationModel.create(asyncOperation);
+
+  const response = await request(app)
+    .post('/asyncOperations')
+    .send(asyncOperation)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(409);
+  const { message } = response.body;
+  t.is(message, (`A DynamoDb record already exists for async operation ID ${asyncOperation.id}`));
+});
+
+test.serial('POST returns a 500 response if record creation throws unexpected error', async (t) => {
+  const { jwtAuthToken } = t.context;
+  const stub = sinon.stub(AsyncOperationModel.prototype, 'create')
+    .callsFake(() => {
+      throw new Error('unexpected error');
+    });
+
+  const asyncOperation = {
+    id: uuidv4(),
+    status: 'RUNNING',
+    taskArn: randomString(),
+    description: 'Some async run',
+    operationType: 'Bulk Granules',
+    output: JSON.stringify({ age: 59 }),
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  };
+
+  const response = await request(app)
+    .post('/asyncOperations')
+    .send(asyncOperation)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(500);
+
+  t.is(response.status, 500);
+  t.teardown(() => {
+    stub.restore();
+  });
+});
+
+test('POST returns a 400 response if invalid record is provided', async (t) => {
+  const { jwtAuthToken } = t.context;
+  const asyncOperation = { invalid: 'value' };
+
+  const response = await request(app)
+    .post('/asyncOperations')
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .send(asyncOperation)
+    .expect(400);
+  t.is(response.status, 400);
+});
+
+test('POST returns a 404 if the requested path does not exist', async (t) => {
+  const { jwtAuthToken } = t.context;
+  const asyncOperation = {
+    id: uuidv4(),
+    status: 'RUNNING',
+    taskArn: randomString(),
+    description: 'Some async run',
+    operationType: 'Bulk Granules',
+    output: JSON.stringify({ age: 59 }),
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  };
+  const response = await request(app)
+    .post(`/asyncOperations/${asyncOperation.id}`)
```

Comment:
```suggestion
    .post(`/asyncOperations/${randomString(}`)
```

Really doing a POST on `asyncOperations/<anything>` would fail with a 404 because that's not a valid route. So I'd prefer this test setup and remove the `asyncOperation` for this test because it's unnecessary

Reply:
Oh that's a good point!
[`e3a0dc2` (#2294)](https://github.com/nasa/cumulus/pull/2294/commits/e3a0dc28f82db01ae42ff09d84d04da7933de9f6)

Reply:
[`f89721f` (#2294)](https://github.com/nasa/cumulus/pull/2294/commits/f89721f1a3c532e130e10c619654f1d4fedf13f2)

apologies this is in two commits